server/src/main/java/org/opensearch/extensions/ExtensionActionListenerHandler.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.extensions;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchException;
+
+/**
+ * Handles ActionListener requests from extensions
+ *
+ * @opensearch.internal
+ */
+public class ExtensionActionListenerHandler {
+
+    private static final Logger logger = LogManager.getLogger(ExtensionActionListener.class);
+    private ExtensionActionListener<ExtensionBooleanResponse> listener;
```

Comment:
It's required to extend the generic ActionListener abstract class/interface

Reply:
I just did this and it seems to work:
```
public class ExtensionActionListener implements ActionListener<Response> {
```
Since you never use whatever type you put there inside the class, it is meaningless.  If you really want a generic type you could do this and it would behave identically:
```
public class ExtensionActionListener<Foo> implements ActionListener<Response> {
```
since the `Foo` type is never used inside that class.

By using the `ExtensionBooleanResponse` name there you are defining a local generic type to be used inside that class, but it's still a placeholder, but confusing since it matches the name of an actual class.  

![image](https://user-images.githubusercontent.com/9291703/191372677-c8ee067d-932e-4329-bb10-de7d17d9ff8c.png)

Reply:
Do you think a better solution is to put `ExtensionActionListener<T>` or what do you recommend?

Reply:
@dbwiddis any thoughts here?

Reply:
I suggest removing the generic type from class `ExtensionActionListener` as the type is never used internally to it so it's meaningless.  Then it can also be removed from this line.